core/controllers/base.py

```diff
@@ -845,6 +847,16 @@ def handle_exception(
             self._render_exception(values)
             return
 
+        if isinstance(exception, TypeError):
+            self.error(405)
+            values = {
+                'error': 'Invalid method %s for %s/ Extra Parameters passed' % (
```

Comment:
Per the HTTP spec, is "extra parameters passed" a valid condition for surfacing 405 errors? If so, could you please point to some documentation to confirm that?

Reply:
I added "extra parameters passed" for the test in `profile_test.py` below, for more context, you can refer this [review](https://github.com/oppia/oppia/pull/20544#discussion_r1650156797).